src/algorithm/butterflies.rs

```diff
@@ -248,60 +246,54 @@ impl IsInverse for Butterfly4 {
 
 
 pub struct Butterfly5<T> {
-	inner_fft_multiply: [Complex<T>; 4],
+    twiddle1: Complex<T>,
+    twiddle2: Complex<T>,
 	inverse: bool,
 }
 impl<T: FFTnum> Butterfly5<T> {
     pub fn new(inverse: bool) -> Self {
-
-    	//we're going to hardcode a raders algorithm of size 5 and an inner FFT of size 4
-    	let quarter: T = FromPrimitive::from_f32(0.25f32).unwrap();
-    	let twiddle1: Complex<T> = twiddles::single_twiddle(1, 5, inverse) * quarter;
-    	let twiddle2: Complex<T> = twiddles::single_twiddle(2, 5, inverse) * quarter;
-
-    	//our primitive root will be 2, and our inverse will be 3. the powers of 3 mod 5 are 1.3.4.2, so we hardcode to use the twiddles in that order
-    	let mut fft_data = [twiddle1, twiddle2.conj(), twiddle1.conj(), twiddle2];
-
-    	let butterfly = Butterfly4::new(inverse);
-    	unsafe { butterfly.process_inplace(&mut fft_data) };
-
+        let twiddle1: Complex<T> = twiddles::single_twiddle(1, 5, inverse);
+        let twiddle2: Complex<T> = twiddles::single_twiddle(2, 5, inverse);
         Butterfly5 { 
-        	inner_fft_multiply: fft_data,
-        	inverse: inverse,
+            twiddle1, 
+            twiddle2,
+        	inverse,
         }
     }
 }
+
 impl<T: FFTnum> FFTButterfly<T> for Butterfly5<T> {
     #[inline(always)]
     unsafe fn process_inplace(&self, buffer: &mut [Complex<T>]) {
-        //we're going to reorder the buffer directly into our scratch vec
-        //our primitive root is 2. the powers of 2 mod 5 are 1, 2,4,3 so use that ordering
-        let mut scratch = [*buffer.get_unchecked(1), *buffer.get_unchecked(2), *buffer.get_unchecked(4), *buffer.get_unchecked(3)];
-
-        //perform the first inner FFT
-        Butterfly4::new(self.inverse).process_inplace(&mut scratch);
-
-        //multiply the fft result with our precomputed data
-        for i in 0..4 {
-            scratch[i] = scratch[i] * self.inner_fft_multiply[i];
-        }
-
-        //perform the second inner FFT
-        Butterfly4::new(!self.inverse).process_inplace(&mut scratch);
-
-        //the first element of the output is the sum of the rest
-        let first_input = *buffer.get_unchecked_mut(0);
-        let mut sum = first_input;
-        for i in 1..5 {
-            sum = sum + *buffer.get_unchecked_mut(i);
-        }
+        let sum = *buffer.get_unchecked(0) + *buffer.get_unchecked(1) + *buffer.get_unchecked(2) + *buffer.get_unchecked(3) + *buffer.get_unchecked(4);
```

Comment:
Could you leave come comments about how this algorithm was derived? I think I know what's happening here -- it looks familiar to me because I recently implemented the same thing in the SIMD branch -- but someone else reading it may not understand.

One of my goals for this project is for this to be something people can refer to when implementing FFTs themselves. I don't think each and every line needs to be commented, but at least linking to a wikipedi page, or telling readers what  algebraic steps will lead them to this set of operations, will go a long way.